### PR TITLE
Ensure DateUtils timezone isn't undefined

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details
Slightly incorrect check in https://github.com/Expensify/Expensify.cash/pull/1573 is leading to whitescreens when you load up E.cash unless you refresh. 

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes Expensify/Expensify/issues/156383

### Tests
1. Sign into a new account on E.cash 
2. Ensure you don't get a blank white screen and that the app loads as anticipated.

### Tested On
- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
N/A

### CODE REVIEW

**Review by Jag96 — APPROVED** [review]
LGTM, feel free to merge once tests finish

File: src/libs/DateUtils.js [hunk 1]
```diff
@@ -10,10 +10,9 @@ let timezone;
 Onyx.connect({
     key: ONYXKEYS.MY_PERSONAL_DETAILS,
     callback: (val) => {
-        if (val) {
-            timezone = _.isObject(val) ? val.timezone.selected : val.timezone;
-        } else {
-            timezone = CONST.DEFAULT_TIME_ZONE.selected;
+        timezone = val ? val.timezone : CONST.DEFAULT_TIME_ZONE.selected;
```
Comment (tgolen) [inline src/libs/DateUtils.js:13]: Hi Nikki, would you mind adding some code comments to explain why the logic needs to be this way? Just looking at the code it is very confusing and could use some context. Also, I am curious if lodash.get() would be helpful at all?

Comment (NikkiWines) [reply src/libs/DateUtils.js:13]: Sure, I can add some comments here

Comment (NikkiWines) [reply src/libs/DateUtils.js:13]: Lodash get could also be helpful here for sure, though I think using a ternary accomplishes the same logic in the same number of lines - open to modifying it though!

Comment (tgolen) [reply src/libs/DateUtils.js:13]: I think the part that is confusing when I look at the code is that there is a ternary followed by an `if`, and I can't really reason about why that is necessary. Maybe if I had some more context around why the logic is this way, I could provide some more helpful suggestions.

Comment (NikkiWines) [reply src/libs/DateUtils.js:13]: For some reason, I can't attach 👍  to your comment but I see your point, and a clarifying comment would definitely be helpful here

**Pull request merged**